src/main/java/frc/robot/subsystems/intakeLimelight/IntakeLimelight.java

```diff
@@ -0,0 +1,575 @@
+   *     camera and positive y is to the left from the camera's point of view.
+   */
+  public Translation2d getTargetTranslation() {
+    return getTargetTranslation(kTargetHeight);
+  }
+
+  /** Gets the object lowest in the image */
+  public Optional<Double> getXOfSmallestY(double[] corners) {
+    // Format: x1 y1 x2 y2 x3 y3 . . .
+    double minY = 10000000;
+    int minYIdx = 0;
+
+    if (corners.length == 0) {
+      return Optional.empty();
+    }
+
+    for (int i = 1; i < corners.length; i += 2) {
+      if (corners[i] < minY) {
+        minY = corners[i];
+        minYIdx = i;
+      }
+    }
+
+    return Optional.of(corners[minYIdx - 1]);
+  }
+
+  public Optional<Double> getObjectHeightPx(double[] corners) {
+    // Format: x1 y1 x2 y2 x3 y3 . . .
+    if (corners.length == 0) {
+      return Optional.empty();
+    }
+
+    double minY = Double.MAX_VALUE;
+    double maxY = Double.MIN_VALUE;
+
+    for (int i = 1; i < corners.length; i += 2) {
+      minY = Math.min(minY, corners[i]);
+      maxY = Math.max(maxY, corners[i]);
+    }
+
+    return Optional.of(maxY - minY);
+  }
+
+  public class NoteDetection {
+    public double latencySec;
+
+    /** Distance from camera */
+    public double distanceMeters;
+
+    /** CCW Yaw Angle */
+    public double yawAngleDeg;
+
+    public NoteDetection(double latencySec, double distanceMeters, double yawAngleDeg) {
+      this.latencySec = latencySec;
+      this.distanceMeters = distanceMeters;
+      this.yawAngleDeg = yawAngleDeg;
+    }
+  }
+
+  /**
+   * Looks for a note
+   *
+   * @return If empty, no note detected. First value is note yaw in degrees (ccw), second value is
+   *     distance in meters.
+   */
+  public Optional<NoteDetection> getNotePos() {
+    if (getPipeline() != pipeline.NOTE_PIPELINE) {
+      setPipeline(pipeline.NOTE_PIPELINE);
+    }
+
+    double[] corners = table.getEntry("tcornxy").getDoubleArray(new double[0]);
+    Optional<Double> maybeXPixels = getXOfSmallestY(corners);
+    Optional<Double> maybeYHeightPixels = getObjectHeightPx(corners);
+
+    if (!maybeXPixels.isPresent() || !maybeYHeightPixels.isPresent() || table.getEntry("tclass").getString(IntakeLimelightConstants.INTAKE_LIMELIGHT_NAME) != "note") {
+      System.out.println("Didn't see note");
+      return Optional.empty();
+    }
+
+    // Compute note distance
+    // kinda based on the same thing as below:
+    // https://docs.limelightvision.io/en/latest/theory.html#from-pixels-to-angles
+    final double vertFovDeg = 49.7;
+    double halfResYPixels = RESOLUTION_Y / 2.0;
+    double viewplaneHeightPixels = 2.0 * Math.tan(Units.degreesToRadians(vertFovDeg / 2));
+    double yPixels = maybeYHeightPixels.get();
+    double normYPixels = (1 / halfResYPixels) * yPixels;
+    double viewplaneYPixels = viewplaneHeightPixels / 2.0 * normYPixels;
+    final double NOTE_HEIGHT_METERS = Units.inchesToMeters(13.0);
```

Comment:
A note is not 13 inches tall lol. It's just above the ground, we can call it 1 inch